### PR TITLE
1952 - ODP previous year selection

### LINKED ISSUE #1952: Original Data Point Page: Copy previous values -> Prefill
Currently, the add national data point dialogue has a button "copy previous values" next to "National classifications". It functions perfectly, but to me if would be more logical to have this button above the " National Classifications" and change the label to "prefill with [year] previous datat" / or just "prefill" as the button prefills both the classifications, and the data and reclassification sections. The [year] would become the year of prev datapoint.

image

Just a thought.

### PR DESCRIPTION
**Pull request by rfontanarosa**
close #1952 

https://user-images.githubusercontent.com/10047945/234849898-9cf7af68-a082-4f05-acf2-bf9d4a74d747.mov

### CODE REVIEW

**Comment by sorja** [comment]
Looks good, should we allow prefilling also with future values now that you can choose the year?

File: src/i18n/resources/en.js [hunk 1]
```diff
@@ -479,7 +479,9 @@ The FRA team
     referenceYearData: 'Reference year for the data',
     referenceYear: 'Reference year',
     references: 'References',
-    selectYear: 'Select…',
+    prefill: 'Prefill',
+    prefillWith: 'Prefill with',
+    selectYear: 'Select year',
```
Comment (sorja) [inline src/i18n/resources/en.js:484]: I would put under 'common'

File: src/meta/assessment/originalDataPoint/odps/getPreviousODPYears.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ * @param years List of reserved ODP years
+ */
+export const getPreviousODPYears = (year: number, years: Array<number>): Array<number> | undefined => {
+  const arr = [...years]
+  // Add and sort current year
+  arr.push(year)
+  arr.sort()
+  // Find position of current year
+  const idx = arr.findIndex((y) => y === year)
+  return idx === 0 ? undefined : arr.slice(0, idx)
```
Comment (sorja) [inline src/meta/assessment/originalDataPoint/odps/getPreviousODPYears.ts]: can this be just `years.filter(y => y < year)`

**Comment by minotogna** [comment]
> Looks good, should we allow prefilling also with future values now that you can choose the year?

actually any year can be selected matching the condition: ODP with classes.length > 0.

the endpoint get years can be updated including this new information. 

what do you think @sorja @rfontanarosa ?

**Comment by sorja** [comment]
> > Looks good, should we allow prefilling also with future values now that you can choose the year?
> 
> actually any year can be selected matching the condition: ODP with classes.length > 0.
> 
> the endpoint get years can be updated including this new information.
> 
> what do you think @sorja @rfontanarosa ?

This makes the most sense, yes

File: src/meta/assessment/originalDataPoint/odps/getPreviousODPYears.ts (hunk 2 above)
Comment (rfontanarosa) [reply src/meta/assessment/originalDataPoint/odps/getPreviousODPYears.ts]: I removed getPreviousODPYears completely :)

File: src/i18n/resources/en.js (hunk 1 above)
Comment (rfontanarosa) [reply src/i18n/resources/en.js:484]: noted!

File: src/client/store/ui/originalDataPoint/actions/getOriginalDataPointReservedYears.ts [hunk 3]
```diff
@@ -3,19 +3,18 @@ import axios from 'axios'
 
 import { ApiEndPoint } from '@meta/api/endpoint'
 import { CycleParams } from '@meta/api/request'
+import { ODPReservedYear } from '@meta/assessment/originalDataPoint'
```
Comment (minotogna) [inline src/client/store/ui/originalDataPoint/actions/getOriginalDataPointReservedYears.ts]: better to import ODPReservedYear from '@meta/assessment'

File: src/client/store/ui/originalDataPoint/stateType.ts [hunk 4]
```diff
@@ -1,7 +1,8 @@
 import { OriginalDataPoint } from '@meta/assessment'
+import { ODPReservedYear } from '@meta/assessment/originalDataPoint'
```
Comment (minotogna) [inline src/client/store/ui/originalDataPoint/stateType.ts]: better to import ODPReservedYear from @meta/assessment

**Review by minotogna — COMMENTED** [review]
good pr 👍  just really some import 😅

**Review by minotogna — APPROVED** [review]

**Review by sorja — APPROVED** [review]

**Pull request merged**